snapcraft.yaml

```diff
@@ -23,45 +23,31 @@ apps:
     command: certbot
     environment:
       PATH: "$SNAP/bin:$SNAP/usr/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games"
-      AUGEAS_LENS_LIB: $SNAP/usr/share/augeas/lenses/dist
+      AUGEAS_LENS_LIB: "$SNAP/usr/share/augeas/lenses/dist"
+      LD_LIBRARY_PATH: "$SNAP/usr/lib/x86_64-linux-gnu/:$LD_LIBRARY_PATH"
   renew:
     command: certbot -q renew
     daemon: oneshot
     environment:
       PATH: "$SNAP/bin:$SNAP/usr/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games"
       AUGEAS_LENS_LIB: $SNAP/usr/share/augeas/lenses/dist
+      LD_LIBRARY_PATH: "$SNAP/usr/lib/x86_64-linux-gnu/:$LD_LIBRARY_PATH"
     passthrough:
         # Run approximately twice a day with randomization
         timer: 00:00~24:00/2
 
 parts:
-  python-augeas:
-    plugin: python
-    source: git://github.com/basak/python-augeas
-    source-branch: snap
-    python-version: python2
-    build-packages: [libaugeas0]
-  certbot:
-    plugin: python
-    source: git://github.com/certbot/certbot
-    source-branch: master
-    python-version: python2
-    after: [python-augeas]
-  certbot-nginx:
+  acme:
     plugin: python
     source: git://github.com/certbot/certbot
-    source-branch: master
-    source-subdir: certbot-nginx
-    build-packages: [libaugeas-dev]
-    stage-packages: [libaugeas0]
-    after: [certbot]
-    python-version: python2
-  certbot-apache:
+    source-branch: v0.35.1
+    source-subdir: acme
+    python-version: python3
+  certbot:
     plugin: python
     source: git://github.com/certbot/certbot
-    source-branch: master
-    source-subdir: certbot-apache
+    source-branch: v0.35.1
     build-packages: [libaugeas-dev]
     stage-packages: [libaugeas0]
-    after: [certbot]
-    python-version: python2
+    python-packages: [acme, python-augeas==0.5.0]
```

Comment:
Why is it necessary to pin this here? Can it use a requirements.txt instead, so this information doesn't have to be in the snapcraft.yaml, and thus prone to breakage as upstream changes?

Reply:
Why it is necessary, it because python-augeas is badly maintained, and any new version has proven to break certbot. So for now we pin it to 0.5.0.

And yes, more generally, we should pin all versions as it is in certbot-auto, because this particular set of dependencies is the most tested in our CI. Here again I am thinking in dynamically construct it in the CI.